fulp_modules/features/antagonists/bloodsuckers/code/vassal/vassal_datum.dm

```diff
@@ -95,10 +95,11 @@
 	owner.current.log_message("has been vassalized by [master.owner.current]!", LOG_ATTACK, color="#960000")
 	/// Give Recuperate Power
 	BuyPower(new /datum/action/cooldown/bloodsucker/recuperate)
-	/// Give Objectives
-	var/datum/objective/vassal_objective/vassal_objective = new
-	vassal_objective.owner = owner
-	objectives += vassal_objective
+	/// Give Objectives if not a particular type of vassal with a unique objective
```

Comment:
Same here about the src stuff. Make a proc called something like ``forge_objectives`` like other antagonists have, and just have the discordant vassal type overwrite it for its own objective.

Reply:
Alright, done!